Assets/Scripts/Enemy/EnemyController.cs

```diff
@@ -1,21 +1,28 @@
 using System.Collections;
 using System.Collections.Generic;
+using Unity.VisualScripting;
 using UnityEngine;
 
-public abstract class EnemyController : CharacterController
+public abstract class EnemyController : CharController
 {
-    public float enemyMoveSpeed = 3f;
+    protected float enemyMoveSpeed = 3f;
+    protected GameObject refToPlayer;
+    protected Rigidbody2D rb;
 
-    protected override void Update()
+    protected override void Start()
     {
-        base.Update();
+        base.Start();
```

Comment:
This is unnecessary. The `Start` method of the parent will be called by the child if it isn't overridden in the child class anyway. You only call this method if you implement a new behavior in the child but still need the behavior implemented in the parent.

Reply:
other implentaitons are used